cli/azd/pkg/infra/provisioning/bicep/prompt.go

```diff
@@ -350,42 +350,61 @@ func (p *BicepProvider) promptForParameter(
 			return nil, fmt.Errorf("parameter '%s' has no allowed values defined", key)
 		}
 
+		// defaultOption enables running with --no-prompt, taking the default value. We use the first option as default.
+		defaultOption := options[0]
```

Comment:
This isn't explicitly called out in the PR description, but I'm guessing we're making an intentional change to have `--no-prompt` always select the "first value". Is the "first value" is guaranteed to be stable across runs?

I would also admit that as a first-time user, this may not be extremely intuitive, but perhaps there are benefits that we are imagining from this change.

Reply:
yeah, this change is trying to enable `--no-prompt` . The current user's feedback is: "why isn't AZD choosing the highlighted value I see during the prompt like in other prompts, like the location parameter?"  

User is aware of how --no-prompt can automate the process by taking whatever selection is displayed in a list. Wondering why that's not the case when not using location parameters

Reply:
I see. Thanks for sharing that perspective for `--no-prompt` to skip most prompts for convenience.

Just to double check on the initial question: is `options[0]` chosen here corresponding to the first value declared in the allowed list?

For example, in `@allowed(['foo', 'bar'])`, is `foo` always chosen? I guess if we do document it as such, it is understandable behavior, and makes `--no-prompt` useful for this specific purpose, so I'm in favor of doing it assuming the order is indeed preserved.